doc/tutorial/search/index.md

```diff
@@ -0,0 +1,258 @@
+You can also pass multiple queries at the same time:
+
+\snippet doc/tutorial/search/small_snippets.cpp multiple_queries
+
+The returned result will be a vector of individual results. Each element refers to the respective query.
+
+\assignment{Assignment 2}
+Search for all exact occurrences of `GCT` in the text from assignment 1.<br>
+Print the number of hits and the positions in a ascending ordering.<br>
+Do the same for the following text collection:
+\code
+std::vector<dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
+                              "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
+                              "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+\endcode
+\endassignment
+
+\solution
+\include doc/tutorial/search/solution2.cpp
+**Expected output:**
+~~~console
+=====   Running on a single text   =====
+There are 3 hits.
+The positions are [1,41,77]
+===== Running on a text collection =====
+There are 3 hits.
+The positions are [(0,1),(1,9),(2,16)]
+~~~
+\endsolution
+
+## Configuration
+
+We offer the following configurations for allowing errors:
+
+With absolute numbers (seqan3::search_cfg::max_error):
+
+- seqan3::search_cfg::total: Maximum number of total errors
+- seqan3::search_cfg::substitution: Maximum number of substitutions
+- seqan3::search_cfg::insertion: Maximum number of insertions
+- seqan3::search_cfg::deletion: Maximum number of deletions
+
+Use seqan3::search_cfg::max_error_rate with the same elements to define error rates \f$\in[0,1]\f$.
+
+To search for either 1 insertion or 1 deletion, you can write:
+
+\snippet doc/tutorial/search/small_snippets.cpp error_search
+
+Note that the error type do not have to sum up to the total of errors allowed:
+~~~cpp
+configuration const cfg = max_error{total{2}, substitution{2}, insertion{1}, deletion{1}};
+~~~
+Allows 2 errors which can be any combination of 2 substituions, 1 insertion and 1 deletion.
+
+Defining only the total will set all error types to the total.<br>
+Defining any of the error types, but no total, will set the total to the sum of all error types.
+
+\assignment{Assignment 3}
+Search for all occurrences of `GCT` in the text from assignment 1.<br>
+Allow up to 1 substitution and print the matching substring of the reference.
+\endassignment
+
+\solution
+\include doc/tutorial/search/solution3.cpp
+**Expected output:**
+~~~console
+There are 12 hits.
+At position 1: GCT
+At position 5: TCT
+At position 12: GAT
+At position 23: GCC
+At position 36: GAT
+At position 41: GCT
+At position 57: ACT
+At position 62: GCC
+At position 75: GCG
+At position 77: GCT
+At position 83: GCA
+At position 85: ACT
+~~~
+\endsolution
+
+### Search modes
+
+Additionally, you can define what kind of hits should be reported:
+
+- seqan3::search_cfg::all: Report all hits that satisfy the requirement.
+- seqan3::search_cfg::best: Report the best hit, i.e. the hit with the lowest edit distance.
```

Comment:
what happens if there are several best hits? which one is chosen? the first?
add a comment